core/permission/access_test.go

```diff
@@ -224,3 +226,46 @@ func (*accessSuite) TestEqualOrGreaterCloudAccessThan(c *gc.C) {
 	c.Check(addmodel.EqualOrGreaterCloudAccessThan(admin), jc.IsFalse)
 	c.Check(admin.EqualOrGreaterCloudAccessThan(addmodel), jc.IsTrue)
 }
+
+var validateObjectTypeTest = []struct {
+	access     permission.Access
+	accessType permission.ObjectType
+	fail       bool
+}{
+	{access: permission.AdminAccess, accessType: permission.Cloud},
+	{access: permission.SuperuserAccess, accessType: permission.Cloud, fail: true},
+	{access: permission.LoginAccess, accessType: permission.Controller},
+	{access: permission.ReadAccess, accessType: permission.Controller, fail: true},
+	{access: permission.ReadAccess, accessType: permission.Model},
+	{access: permission.ConsumeAccess, accessType: permission.Model, fail: true},
+	{access: permission.ConsumeAccess, accessType: permission.Offer},
+	{access: permission.AddModelAccess, accessType: permission.Offer, fail: true},
+	{access: permission.AddModelAccess, accessType: 5, fail: true},
+}
+
+func (*accessSuite) TestValidateAccessForObjectType(c *gc.C) {
+	size := len(validateObjectTypeTest)
+	for i, test := range validateObjectTypeTest {
+		c.Logf("Running test %d of %d", i, size)
+		id := permission.ID{ObjectType: test.accessType}
+		err := id.ValidateAccess(test.access)
+		if test.fail {
+			c.Check(errors.Is(err, errors.NotValid), jc.IsTrue, gc.Commentf("test %d", i))
```

Comment:
Small nit and don't worry about changing it if you don't want. Simon wrote a jc helper `c.Check(err, jc.ErrorIs, errors.NotValid)`

Reply:
oh, cool, I didn't know that existed, thx